### ISSUE TITLE
Config repo tag: Attributes changed

### ISSUE THREAD

**Issue by arvindsv** [gocd-contrib/gomatic#40]
In GoCD 17.9 (coming next week), the `<config-repo/>` tag has changed a bit. The changes are explained https://github.com/gocd/gocd/issues/3648 and https://github.com/gocd/gocd/issues/3671.

I'm not sure how you handle different versions, but for 17.9 at least, something around [this](https://github.com/gocd-contrib/gomatic/blob/576723a48cfb0b2e016d5189af825fe2aca84bf4/gomatic/gocd/config_repos.py#L64) will probably have to change.

From what I can see in [the GoCD code](https://github.com/gocd/gocd/blob/2854287be009deaa6c8ea6f646eea7b09900f8cb/config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java#L54), the ID gets generated automatically, if not provided.

/cc @dudadornelles

**Comment by dudadornelles** [comment]
Thanks for the heads up.

As for dealing with versions, I'll double check but I didn't see anything in the codebase that deals with them. I'm thinking we could have the version being a parameter for whatever `ensure` method might have its behavior dependent on version or we could potentially get some metadata from the server url after you create the go configurator and make it available in the configurator.

1st option
```
def ensure_config_repo(self, url, plugin, cvs='git', configuration=None, version=None):
    # do something if version and version > x else do something else
```

2nd option
def ensure_config_repo(self, url, plugin, cvs='git', configuration=None):
    version = self.__configurator.version  # i believe every element has access to the configurator
    # do something if version and version > x else do something else
```

**Comment by arvindsv** [comment]
Makes sense. Hopefully, there won't be too many of these changes. I'll try and let you know whenever I see one. Thanks!

**Comment by dudadornelles** [comment]
there is an endpoint that tells the version: https://api.gocd.org/current/#the-version-object

**Comment by dudadornelles** [comment]
Published in 0.5.7!
